lms/djangoapps/ccx/views.py

```diff
@@ -71,43 +71,60 @@ def wrapper(request, course_id):
         and modifying the view's call signature.
         """
         course_key = CourseKey.from_string(course_id)
+        ccx = None
+        if isinstance(course_key, CCXLocator):
+            # is there a security leak here in not checking that this user is
+            # the coach for this ccx?
+            ccx_id = course_key.ccx
+            ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
+            course_key = ccx.course_id
+
         role = CourseCcxCoachRole(course_key)
         if not role.has_user(request.user):
             return HttpResponseForbidden(
                 _('You must be a CCX Coach to access this view.'))
+
         course = get_course_by_id(course_key, depth=None)
-        return view(request, course)
+        return view(request, course, ccx)
     return wrapper
 
 
 @ensure_csrf_cookie
 @cache_control(no_cache=True, no_store=True, must_revalidate=True)
 @coach_dashboard
-def dashboard(request, course):
+def dashboard(request, course, ccx=None):
```

Comment:
@cpennington:  The two places where `ccx=None` makes sense are the `dashboard` view and the `create_ccx` view.  Since we are wrapping all the views in the same decorator, in order to handle ccx and course lookup and access control from a single location, it makes sense to allow the ccx to be `None` when needed.  

We could do a class-based view, I suppose, but I'm really not a big fan of them.  Especially for simple, direct views like this.  It's a lot of magic and indirection added for what I would consider to be minimal benefit.  I'm just old-school like that I guess.